onnx/shape_inference/implementation.h

```diff
@@ -459,6 +459,16 @@ void InferShapes(
     const ISchemaRegistry* schema_registry = OpSchemaRegistry::Instance(),
     const ShapeInferenceOptions& options = {}
     );
+///
+/// Forces shape inference to run with data propagation and data propagation result will be updated in place
```

Comment:
Would it make sense to then just extend the previous method with a 
```cpp
   std::unordered_map<std::string, TensorShapeProto>* generated_shape_data_by_name = nullptr
```
?

Reply:
Thank you for the great idea! Yes, I think we can simply extend existing `InferShapes` instead of introducing more similar API. I have just removed new `InferShapesAndDataPropagation` and use generated_shape_data_by_name as pointer globally. Please review it again.